assets/javascripts/discourse/components/ai-post-helper-menu.gjs

```diff
@@ -0,0 +1,367 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { inject as service } from "@ember/service";
+import { and } from "truth-helpers";
+import CookText from "discourse/components/cook-text";
+import DButton from "discourse/components/d-button";
+import FastEdit from "discourse/components/fast-edit";
+import FastEditModal from "discourse/components/modal/fast-edit";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { sanitize } from "discourse/lib/text";
+import { clipboardCopy } from "discourse/lib/utilities";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+import I18n from "discourse-i18n";
+import eq from "truth-helpers/helpers/eq";
+import AiHelperCustomPrompt from "../components/ai-helper-custom-prompt";
+import AiHelperLoading from "../components/ai-helper-loading";
+
+export default class AiPostHelperMenu extends Component {
```

Comment:
This looks fine to me, but I wonder if some of the logic behind like which request to do can be abstracted into different objects.